### DISCUSSION TITLE
Getting the type for a case class field

### DISCUSSION THREAD

**Discussion by adamw** [scala/scala3#15148] [Metaprogramming] [answered]
I'm trying to get the `TypeRepr` for a `Symbol`, obtained via `Symbol.caseFields`. However, using `Symbol.tree` is discouraged as an anti-pattern, and getting the `TypeRepr` from a `Tree` requires partial pattern matches such as:

```scala
val fieldTypeRepr = field.tree match
  case v: ValDef => v.tpt.tpe
  case d: DefDef => d.returnTpt.tpe
```

What would be the correct way to do this?

**Comment by adamw** [comment, marked as answer]
Assuming that `tpe` is the `TypeRepr` for the case class, field symbols can be accessed using `tpe.typeSymbol.caseFields`. Then, to obtain the `TypeRepr` for a field, you can use `.memberType`, as follows:

```scala
tpe.typeSymbol.caseFields.map { fieldSymbol =>
  val fieldTpe = tpe.memberType(fieldSymbol)
  ...
}
```

**Reply by adamw** [reply]
Feel free to delete the above, I'm not able to, not to create confusion :)

**Reply by bishabosha** [reply]
you could edit your posts so it looks more like an FAQ?

**Reply by adamw** [reply]
@bishabosha Done :)
